visitors/http_rest_diff/http_rest_spec_diff_visitor.go

```diff
@@ -0,0 +1,550 @@
+package http_rest_diff
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+
+	pb "github.com/akitasoftware/akita-ir/go/api_spec"
+	"github.com/akitasoftware/akita-libs/spec_util"
+	. "github.com/akitasoftware/akita-libs/visitors"
+	"github.com/akitasoftware/akita-libs/visitors/go_ast_pair"
+	"github.com/akitasoftware/akita-libs/visitors/http_rest"
+)
+
+// An HttpRestSpecPairVisitor with hooks for processing each difference found
+// between two IR trees. A node is considered changed if a difference can be
+// observed at that level of the IR. For example, HTTPAuth nodes with different
+// Types are considered changed, but their parents might not necessarily be
+// considered changed.
+//
+// Go lacks virtual functions, so all functions here take the visitor itself as
+// an argument, and call functions on that instance.
+type HttpRestSpecDiffVisitor interface {
```

Comment:
Because http_rest_diff is already in the name of the package, I think it would be more idiomatic to use shorter names for the interfaces.  I guess it's too late to do that with http_rest but you can see further down how 

http_rest.SpecPairVisitorContext

conveys the same information as

http_rest.HttpRestSpecPairVisitorContext

Reply:
In 8147936.